cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -73,6 +73,246 @@
 CLUSTER_NAME_AND_CUSTOM_DOMAIN_NAME_MAX_LENGTH = 255 - HOST_NAME_MAX_LENGTH - 1
 
 
+class FlexibleInstanceTypesValidator(Validator):
+    """Validator for Compute Resources that has multiple instance types."""
+
+    def _validate(
+        self,
+        queue_name: str,
+        capacity_type: Enum,
+        allocation_strategy: Enum,
+        compute_resource_name: str,
+        instance_types_info: Dict[str, InstanceTypeInfo],
+        disable_simultaneous_multithreading: bool,
+        efa_enabled: bool,
+        placement_group_enabled: bool,
+        memory_scheduling_enabled: bool,
+    ):
+        self.validate_cpu_requirements(compute_resource_name, instance_types_info, disable_simultaneous_multithreading)
```

Comment:
I'm worried about putting all the checks in a single validator. This because if user want to disable validator to bypass cpu check, then all the other checks will be bypassed as well. I'm rather on the opinion that the check needs to be disabled on demand, one by one. Let's follow up on this.